closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -23,27 +23,53 @@
  */
 
 #include <windows.h>
+#include <stdio.h>
+#include <jni.h>
+#include <sys/stat.h>
 
 #include "NativeCrypto_md.h"
 
+#define JAVA_DLL "java.dll"
+
+/* Prototypes. */
+jboolean GetJREPath(char *path, jint pathsize);
+jboolean GetApplicationHome(char *buf, jint bufsize);
+int JLI_Snprintf(char* buffer, size_t size, const char* format, ...);
+
 /* Load the crypto library (return NULL on error) */
 void * load_crypto_library() {
     void * result = NULL;
     const char *libname;
     const char *oldname = "libeay32.dll";
+    char opensslpath[MAX_PATH];
+    char libpathname[MAX_PATH];
+    int rc;
+    struct stat s;
 
 #if defined (_WIN64)
     libname = "libcrypto-1_1-x64.dll";
 #else
     libname = "libcrypto-1_1.dll";
 #endif
 
-    result = LoadLibrary(libname);
-    
-    if (result == NULL) {
-        result = LoadLibrary(oldname);
-    }
+    if (GetJREPath(opensslpath, MAX_PATH)) {
+        rc = JLI_Snprintf(libpathname, sizeof(libpathname), "%s\\bin\\%s", opensslpath, libname);
+        if (rc > 0 && rc <= MAX_PATH && (stat(libpathname, &s) == 0)) {
```

Comment:
Please add parentheses around the parts of this compound condition (and the similar ones below).

Reply:
Done